### PR TITLE
feat: impl a new ttl based LRU cache & fix issues

### PR DESCRIPTION
**Pull request by Saviio**
WHAT's changed: 

1. Implemented a new TTL-based LRU cache. (currently, some api response will be cached with an expiry time of 3 or 6 hours)
2. Added the __safe_invoke function to doubanapi/apiv2.py. 
3. Separated the calling methods within doubanapi according to API characteristics.
4. Utilized ttl_cached_request in the TMDB API class instead of the original cached_request for fetching remote data.

REASONs for these modifications:

1. Consideration was given to the fact that some data exhibits periodicity, e.g., douban Now Playing , TMDB Trending, etc. However, relying solely on LRU cache behavior would result in the inability to reflect changes in this data properly, both in web UI and web API. In this case, UI or  API have to wait for cached data to be evicted to obtain updated response. Therefore, in this pull request, a naive TTL (time to live)-based LRU cache was implemented to ensure data expiration within an expected time range.

2. Fixed the issue of unable to view resource details on the UI side due to various reasons causing failure in calling doubanapi, such as 429 rate limit. Actively throwing exceptions in the __invoke side ensures that erroneous responses are not cached by the LRU, while maintaining consistent function behavior with the original through __safe_invoke.

3. For TMDB, defaulting to TTL-based requests was chosen. Considering TMDB's relatively lax rate limiting measures, a 6-hour expiry time ensures data freshness without incurring excessive overhead.

### CODE REVIEW

File: app/media/doubanapi/apiv2.py [hunk 1]
```diff
@@ -176,84 +177,122 @@ def __invoke(cls, url, **kwargs):
...
+    def __safe_invoke(self, *args, **kwargs):
         try:
-            return json.loads(resp_json_str)
-        except json.JSONDecodeError as e:
+            return self.__invoke(*args, **kwargs)
+        except Exception as e:
             return {}
 
+    @classmethod
+    @ttl_lru(seconds=60 * 60 * 3, maxsize=256)
```
Comment (Saviio) [inline app/media/doubanapi/apiv2.py:200]: need suggestions to adjust

File: app/media/doubanapi/apiv2.py
Comment (hsuyelin) [inline app/media/doubanapi/apiv2.py:1]: def __safe_invoke(self, *args, **kwargs):
        try:
            return self.__invoke(*args, **kwargs)
        except Exception as e:
            return {}

maybe mistake

Comment (Saviio) [reply app/media/doubanapi/apiv2.py:1]: expain it? What kind of mistake

Comment (hsuyelin) [reply app/media/doubanapi/apiv2.py:1]: > safe_invoke(self, *args, **kwargs):

sorry, I misread the word. It's good.

**Pull request merged**